crates/driver/src/domain/competition/quote.rs

```diff
@@ -0,0 +1,141 @@
+use crate::{
+    domain::{
+        competition::{self, order, solution},
+        eth,
+    },
+    infra::solver::{self, Solver},
+    util::{self, conv},
+};
+
+#[derive(Debug)]
+pub struct Config {
+    pub timeout: solution::SolverTimeout,
+}
+
+/// A quote describing the expected outcome of an order.
+#[derive(Debug)]
+pub struct Quote {
+    /// The amount that can be bought if this was a sell order, or sold if this
+    /// was a buy order.
+    pub amount: eth::U256,
+    pub interactions: Vec<competition::solution::Interaction>,
+}
+
+impl Quote {
+    fn new(order: &Order, solution: competition::Solution) -> Result<Self, Error> {
+        let sell_price = solution
+            .prices
+            .get(&order.sell_token)
+            .ok_or(Error::QuotingFailed)?
+            .to_owned();
+        let buy_price = solution
+            .prices
+            .get(&order.buy_token)
+            .ok_or(Error::QuotingFailed)?
+            .to_owned();
+        let amount = match order.side {
+            order::Side::Sell => conv::u256::from_big_rational(
+                &(conv::u256::to_big_rational(buy_price) / conv::u256::to_big_rational(sell_price)
+                    * conv::u256::to_big_rational(order.amount.into())),
+            ),
+            order::Side::Buy => conv::u256::from_big_rational(
+                &(conv::u256::to_big_rational(sell_price) / conv::u256::to_big_rational(buy_price)
+                    * conv::u256::to_big_rational(order.amount.into())),
+            ),
```

Comment:
I _think_ the exchange rates are inverted. Specifically, `sell_amount * sell_price = buy_amount * buy_price`.

See https://github.com/cowprotocol/contracts/blob/d043b0bfac7a09463c74dfe1613d0612744ed91c/src/contracts/GPv2Settlement.sol#L353-L368

```suggestion
            order::Side::Sell => conv::u256::from_big_rational(
                conv::u256::to_big_rational(order.amount.into()) * conv::u256::to_big_rational(sell_price) / conv::u256::to_big_rational(buy_price),
            ),
            order::Side::Buy => conv::u256::from_big_rational(
                conv::u256::to_big_rational(order.amount.into()) * conv::u256::to_big_rational(buy_price) / conv::u256::to_big_rational(sell_price),
            ),
```

I'm only 90% sure, because price math are like USB sticks, you need to turn it 3 times before it goes in.